genesis/src/vm.rs

```diff
@@ -3,25 +3,61 @@ use {
     aptos_gas_schedule::{LATEST_GAS_FEATURE_VERSION, MiscGasParameters, NativeGasParameters},
     aptos_native_interface::SafeNativeBuilder,
     aptos_types::on_chain_config::{Features, TimedFeaturesBuilder},
-    aptos_vm::natives::aptos_natives_with_builder,
+    aptos_vm_environment::natives::aptos_natives_with_builder,
     move_binary_format::errors::VMError,
-    move_vm_runtime::move_vm::MoveVM,
+    move_vm_runtime::{
+        RuntimeEnvironment, WithRuntimeEnvironment, config::VMConfig, move_vm::MoveVM,
+    },
 };
 
-pub struct MovedVm;
+pub struct MovedVm {
+    env: RuntimeEnvironment,
+}
```

Comment:
Would extending it to contain the `ResolverBasedBytesStorage` make sense here? This would shave a few lines off the preamble we have anywhere VM-related. And the VM would still be decoupled as it' spawned through the trait. And as another note, the naming has become a bit imprecise. Maybe call it `VMContext` or `VMRuntimeContainer` or the like?